lib/jekyll/regenerator.rb

```diff
@@ -94,23 +90,7 @@ def modified?(path)
         return cache[path]
       end
 
-      # Check path that exists in metadata
-      data = metadata[path]
-      if data
-        data["deps"].each do |dependency|
-          if modified?(dependency)
-            return cache[dependency] = cache[path] = true
-          end
-        end
-        if File.exist?(path) && data["mtime"].eql?(File.mtime(path))
-          return cache[path] = false
-        else
-          return add(path)
-        end
-      end
-
-      # Path does not exist in metadata, add it
-      return add(path)
+      check_path_exists(path)
```

Comment:
The comment on the old line 97 only referenced the old line 98. 99 thru 110 determined if the file was modified, so it has nothing to do with the comment. This is three operations:

1. If we have seen this file before and one of its dependencies have been modified, set the regeneration bit for both the dependency and the file to `true`
2. If we have seen this file before and it has not been modified, set the regeneration bit to `false`. If it has been modified, set it to `true`.
3. If we have not seen this file before, add it to the metadata and regenerate it.